Makefile

```diff
@@ -14,8 +14,11 @@ PFM_LIBDIR ?= $(PFM_DIR)/lib
 
 GIT_VERSION := $(shell git describe --dirty --always)
 
+ifneq ($(CPU_ARCH),)
+ARCH_FLAGS := -march=$(CPU_ARCH)
+endif
 O_LEVEL ?= -O2
-CPPFLAGS := -MMD -Wall -g $(O_LEVEL) -march=haswell -DGIT_VERSION=\"$(GIT_VERSION)\" -DUSE_LIBPFC=$(USE_LIBPFC) \
```

Comment:
Yeah I added this because `-march=native` works kind of stupidly on gcc: if your CPU is new enough that gcc doesn't recognize it, it just silently ignores the `-march` flag and falls back to some quite old x86 model which doesn't even try to fuse test/jmp in loops, which makes some types of benchmarks less reliable. 

... but yeah, hardcoding `-march=haswell` isn't really the right thing to do for a general purpose test. I kind of wanted to have a smarter approach to detecting the `-march`, but it doesn't seem super easy (see [here](https://stackoverflow.com/q/47299458/149138) for a discussion of how to event detect which options are available).

I might still set the default `CPU_ARCH` to Haswell in `config.mk`, but you can always override in your `local.mk`.